src/app/shared/popup-notification.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PopupNotificationService {
+
+  constructor(private snackBar: MatSnackBar) {
+  }
+
+  showSuccessMessage(message: string) {
+    this.openSnackBar(message, 'success-notification');
+  }
+
+  showWarningMessage(message: string) {
+    this.openSnackBar(message, 'warning-notification');
+  }
+
+  showErrorMessage(message: string) {
+    this.openSnackBar(message, 'error-notification');
+  }
+
+  private openSnackBar(message: string, className: string) {
```

Comment:
Wouldn't it be better to restrict `className` with the aid of some enumeration like `className: POPUP_NOTIFICATION_CLASSNAME` that maps to an actual classname?

Reply:
Good point. Thank you!